pkg/oc/cli/cmd/idle.go

```diff
@@ -363,33 +331,26 @@ func findScalableResourcesForEndpoints(endpoints *kapi.Endpoints, decoder runtim
 	}
 
 	// ... then, for each pod, we check the controller, and find the set of unique controllers...
-	immediateControllerRefs := make(map[kapi.ObjectReference]struct{})
+	immediateControllerRefs := make(map[metav1.OwnerReference]struct{})
 	for _, pod := range podRefs {
-		controllerRef, err := getControllerRef(pod, decoder)
-		if err != nil {
-			return nil, fmt.Errorf("unable to find controller for pod %s/%s: %v", pod.Namespace, pod.Name, err)
-		} else if controllerRef == nil {
+		controllerRef := metav1.GetControllerOf(pod)
+		if controllerRef == nil {
 			return nil, fmt.Errorf("unable to find controller for pod %s/%s: no creator reference listed", pod.Namespace, pod.Name)
 		}
-
 		immediateControllerRefs[*controllerRef] = struct{}{}
 	}
 
 	// ... finally, for each controller, we load it, and see if there is a corresponding owner (to cover cases like DCs, Deployments, etc)
 	controllerRefs := make(map[unidlingapi.CrossGroupObjectReference]struct{})
 	for controllerRef := range immediateControllerRefs {
-		controller, err := getController(controllerRef)
+		controller, err := getController(endpoints.Namespace, controllerRef)
 		if utilerrors.TolerateNotFoundError(err) != nil {
 			return nil, fmt.Errorf("unable to load %s %q: %v", controllerRef.Kind, controllerRef.Name, err)
 		}
 
 		if controller != nil {
-			var parentControllerRef *kapi.ObjectReference
-			parentControllerRef, err = getControllerRef(controller, decoder)
-			if err != nil {
-				return nil, fmt.Errorf("unable to load the creator of %s %q: %v", controllerRef.Kind, controllerRef.Name, err)
-			}
-
+			var parentControllerRef *metav1.OwnerReference
```

Comment:
Next time `parentControllerRef := metav1.GetControllerOf(controller)` you'll save yourself a few precious key strokes 😉

Reply:
lol, yeah, I think that was left over from before I refactored it a bit, and I needed the pre-assignment declaration, and never simplified after